src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy

```diff
@@ -21,6 +21,27 @@ import org.gradle.testkit.runner.TaskOutcome
 
 class DockerCreateContainerFunctionalTest extends AbstractGroovyDslFunctionalTest {
 
+    def "can setup additional user groups"() {
+        given:
+        String containerCreationTask = """
+            task createContainer(type: DockerCreateContainer) {
+                dependsOn pullImage
+                targetImageId pullImage.getImageId()
+                cmd = ['groups']
+                groups = ['postgres']
+            }
+        """
+        buildFile <<
+            containerStart(containerCreationTask) <<
+            containerLogAndRemove()
+
+        when:
+        BuildResult result = build('logContainer')
+
+        then:
+        result.output.contains("postgres")
```

Comment:
Is there a little bit more output we could verify?

Reply:
> Can we do something more specific like print out the groups of the given user and ensure a group which the user should not have is added and present?

This is what this test do right now. I just wonder if we actually should test Docker here (whether it add groups or not) or maybe we should just inspect container and see if additional groups are returned - WDYT?

Reply:
Inspecting the container is the way to go IMHO.